pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/PulsarAuthorizationProvider.java

```diff
@@ -641,4 +643,131 @@ public CompletableFuture<Boolean> validateTenantAdminAccess(String tenantName, S
                 });
     }
 
+    @Override
+    public CompletableFuture<Void> removePermissionsAsync(TopicName topicName) {
+        return getPoliciesReadOnlyAsync().thenCompose(readonly -> {
+            if (readonly) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Policies are read-only. Broker cannot do read-write operations");
+                }
+                throw new IllegalStateException("policies are in readonly mode");
+            }
+            return pulsarResources.getNamespaceResources().getPoliciesAsync(topicName.getNamespaceObject())
+                    .thenCompose(policies -> {
+                        if (!policies.isPresent()
+                                || !policies.get().auth_policies.getTopicAuthentication()
```

Comment:
Does `auth_policies` can be null?

Reply:
It won't, see https://github.com/apache/pulsar/blob/4586852f30902b5247d907ce4cc9360152db1e21/pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/Policies.java#L33-L37